Restructure file system

ItsDrike opened the pull request:
## Overview

We need to update our filestructure to a more scalable model.
Right now the file structure is getting very big and it's unpractical.

## TODO

* [x] Move every bot code related file into `bot/` directory
    - [x] Optimize imports to handle this
* [x] Rename `main.py` to `__main__.py` in order to be able to start it as a module
* [x] Merge all math-related files into single `cogs/math.py`
    - [x] Merge `octcalc.py`, `hexcalc.py` and `bincalc.py`
* [x] Remove `embed_handler.py` with only 1 function in it, referenced only once throughout the project
* [x] Remove `members.py` with only 1 function in it, referenced only once throughout the project
* [x] Remove unused `conversion.py`
* [x] Merge `cogs/utils` into `utils`
* [x] Merge `urbandictpages.py` into `study.py`
* [x] Move `constants.py` into `bot/`
* [x] Rename `constants.py` to `config.py`
* [x] Use `assets/filtered_words.txt` instead of defining these in python file
* [x] Rename `code_utils` to `coding` and use it as a module wiht `__init__.py` for cog loading and `coding.py` as the cog itself
* [x] Standardize hangman to use a separate class instead of defining everything in 1 command function
* [x] Split games cog into games/ directory which will contain games.py for smaller games and hangman.py for hangman
* [x] Use .txt file to store `allowed_filetypes` (AntiMalware) and `filter_words` instead of python files
* [x] Enable `malware_protection` cog
* [x] Rename `malware_protection.py` cog to `security.py`
* [x] Remove `search_utils/` and merge it's content directly into `search.py` cog
* [x] Remove `endpoints` and add a single dict that it contains into `constants.py`
* [x] Remove `emote_utils/` and merge its content directly into `emotes.py` cog
* [x] Split `coding.py` into `coding.py` and `documentation.py`
* [x] Optimize `coding.py`
* [x] Use proper imports for Bot child class from `bot/core/bot` instead of using `discord.ext.commands.Bot` for type hints
* [x] Move `assets/checks.py` to `utils/checks.py`
* [x] Create `core/converters.py` which will be used to store all converters in the project
* [x] Create `core/decorators.py` which will be used to store all decorators in the project
* [x] Cleanup `bot/utils/errors.py`
* [x] Rename `converters.py` cog to `conversion.py` to avoid confustion with `bot/core/converters.py`
* [x] Remove redundant `infog.py` cog

## Additional Notes


* This will affect a lot of other PRs and while Git will help you a lot with resolving PRs since it understands moving files around, it might still cause a lot of problems on merging with other PRs since files weren't only moved but also splitted/merged or removed, which (obviously) git won't help you with.
* Before this can be added we need to make sure everything works, that means testing every single function and since we don't have any automated unit tests built, for now, we'll need to test everything manually.

ItsDrike committed:
Move bot files into bot directory

ItsDrike committed:
Merge calculators into single file

ItsDrike committed:
Merge all math-related files

ItsDrike committed:
Move constants into bot

ItsDrike committed:
remove unused conversion.py

ItsDrike committed:
Remove embed handler and members.py
- These files are only being used in commands, keeping them there is good enough

janaSunrise reviewed (commented):
Change `constants.py` to `config.py`

ItsDrike committed:
Merge urbandictpages into study.py

ItsDrike committed:
Merge cogs/utils into utils

ItsDrike committed:
Fix search cog, use filter_words as txt file

ItsDrike committed:
Split games cog into games/ dir

Further changes:
- Edit Hangman to comply with code standards
- Edit flake8 config to ignore classmethod annotations

ItsDrike committed:
Update start script

- Respect new filesystem for start script in Pipfile

ItsDrike committed:
Merge branch 'master' into update-filestructure

ItsDrike committed:
Use txt file to store allowed filetypes

ItsDrike committed:
Remove emote_utils/ directory

- This directory was only used to store exception and a single class
  that is unacceptable since this class wasn't even referenced anywhere
  outside of Emotes cog, so there is no point in having a separate file
  for it.

ItsDrike committed:
Improve emotes cog readability

ItsDrike committed:
Remove search_utils

- Search utils was only used by search cog, which means it doesn't need
  a separate file.
- search_utils directory only contained 1 file (searchexception) which
  only contained 1 single exception (SafesearchFail), it doesn't make
  sense to use a separate file only for this

ItsDrike committed:
Remove endpoints

-Endpoints only contained 1 dictionary with a signle element, this was
moved to constants

ItsDrike committed:
Add core/bot.py for Bot clas

- It makes sense to separate the Bot class from __main__.py as it will
  be getting imported from many other files due to type hints
- This also makes __main__.py much more clean and readable

ItsDrike committed:
Add extension loading

ItsDrike committed:
Fix malware protection filter

- Whitelist was loading `\n` character when loading the whitelist which
  caused it to match all given extensions.

ItsDrike committed:
Remove core/logging.py which was only a placeholder

- This will be taken care of in another PR, there is no need to have
  this here only as a placeholder

ItsDrike committed:
Remove redundant print statement

ItsDrike committed:
Add embeds cog to extension list

ItsDrike committed:
Update log channel ID

ItsDrike committed:
Merge branch 'master' into update-filestructure

janaSunrise commented on bot/assets/allowed_filetypes.txt:
```diff
@@ -0,0 +1,28 @@
+.tiff
+.wmv
+.gif
+# Photos
```
Doesn't these comments get read by file reader?

ItsDrike replied:
No, there is a handler especially for that https://github.com/The-Codin-Hole/HotWired-Bot/blob/0a45e395fa0b8efab8548734d2ad5762de4b815c/bot/cogs/malware_protection.py#L22-L26

janaSunrise commented on bot/assets/filter_words.txt:
```diff
@@ -0,0 +1,96 @@
+tranny
+\bfa+g(got)?
```
Are these read as regexes?

ItsDrike replied:
Yes, fairly obvious from code in `search.py` https://github.com/The-Codin-Hole/HotWired-Bot/blob/0a45e395fa0b8efab8548734d2ad5762de4b815c/bot/cogs/search.py#L12-L21

janaSunrise commented on bot/cogs/commands.py:
```diff
@@ -0,0 +1,175 @@
+                Bots: {bots}
+            """
+        status = f"""
+                <:online:346921745279746048> {member_by_status["online"]}
+                <:away:346921747330891780> {member_by_status["idle"]}
+                <:dnd:346921781786836992> {member_by_status["dnd"]}
+                <:offline:346921814435430400> {member_by_status["offline"]}
```
```suggestion
                STATUSES["Status.online"] {member_by_status["online"]}
                STATUSES["Status.idle"] {member_by_status["idle"]}
                STATUSES["Status.dnd"] {member_by_status["dnd"]}
                STATUSES["Status.offline"] {member_by_status["offline"]}
```

janaSunrise commented on bot/cogs/games/hangman.py:
```diff
@@ -0,0 +1,193 @@
+
+from bot import constants
+
+DIR_PATH = path.dirname(__file__)
```
What's this, i don't know about this.

ItsDrike replied:
It's just a DIR_PATH constant, so I don't need to use `path.dirname(__file__)` to get the name of the directory `hangman.py` is in, in this case, it will return `bot/cogs/games`, if it were moved to some other directory, it would change to that.

ItsDrike committed:
Install bleach and update lockfile

ItsDrike committed:
Basic coding.py split

- coding.py was split into coding.py and documentation.py because it was
  too big
- This is just a first split commit, there will likely be more
- This also adjusts man command functionality

ItsDrike committed:
Remove commands moved to documentation.py

ItsDrike committed:
Update is_owner check position

ItsDrike committed:
Rename malware_protection to security

-Cog name malware_protection is quite long and security seems like a
more appropriate name for it

ItsDrike committed:
Update lockfile

ItsDrike committed:
Merge branch 'master' into update-filestructure

ItsDrike committed:
Use single directory for `coding.py`

ItsDrike committed:
Make sure "pool" attribute exist before closing

ItsDrike committed:
Rename `constants.py` to `config.py`

ItsDrike committed:
Use proper Bot from `bot/core/bot` instead of `ext.commands.Bot`

ItsDrike committed:
Use `nekos` instead of `n`

ItsDrike committed:
Fix bugs in fun.py

ItsDrike committed:
Move checks to utils/

ItsDrike committed:
Move all converters into converters.py

ItsDrike committed:
Move all decorators to `decorators.py`

ItsDrike committed:
Cleanup `errors.py`

ItsDrike committed:
Rename converters cog to conversion

- To avoid confusion between `bot/core/converters.py` and `bot/cogs/converters.py`

ItsDrike committed:
Remove infog cog

ItsDrike committed:
Use status on members from STATUSES

janaSunrise commented on bot/cogs/codesandbox.py:
```diff
@@ -1,12 +1,14 @@
 
 from discord import Forbidden, HTTPException
-from discord.ext.commands import Bot, Cog, Context, command, is_owner, CommandError
+from discord.ext.commands import Cog, CommandError, Context, command, is_owner
```
```suggestion
from discord.ext.commands import (
        Cog, 
        CommandError, 
        Context, 
        command, 
        is_owner
)
```

ItsDrike replied:
No, this was automatically formatted by `flake8` standards, there is no reason to split imports like this
